### PR TITLE
:sparkles: feat: generate json function

### PR DESCRIPTION
**Pull request by joschkabraun**
By only generating a function which receives JSON and returns a JSON we can solve different issues:
- GPT learning about intricacies of `docarray` and `jina`
  - esp. clashes of `jina.requests` and `requests` package
- we need less iterations as writing the code as become more precise
- context length: we can drop a lot of the system messages as they were necessary for `docarray` and `jina`
  - solves #41 
  - solves #62

### CODE REVIEW

File: src/options/generate/generator.py [hunk 1]
```diff
@@ -76,7 +77,7 @@ def generate_and_persist_file(
             destination_folder: str,
             file_name_s: Union[str, List[str]] = None,
```
Comment (joschkabraun) [inline src/options/generate/generator.py:78]: as discussed in pairing, let's think more about this variable

File: src/options/generate/generator.py [hunk 2]
```diff
@@ -124,18 +125,26 @@ def generate_microservice(
...
+
         microservice_content = self.generate_and_persist_file(
-            'Microservice',
-            template_generate_executor,
-            MICROSERVICE_FOLDER_v1,
-            microservice_name=microservice_name,
+            section_title='Microservice',
+            template=template_generate_function,
+            destination_folder=MICROSERVICE_FOLDER_v1,
+            # microservice_name=microservice_name,
```
Comment (joschkabraun) [inline src/options/generate/generator.py:140]: remove commented code

File: src/options/generate/templates_system.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 from src.constants import FLOW_URL_PLACEHOLDER
-from src.options.generate.templates_user import not_allowed_docker_string, not_allowed_executor_string
+from src.options.generate.templates_user import not_allowed_docker_string, not_allowed_function_string
```
Comment (joschkabraun) [inline src/options/generate/templates_system.py:2]: double check to actually remove the system messages

File: src/options/generate/templates_user.py [hunk 4]
```diff
@@ -91,49 +87,33 @@
 You must provide the complete file with the exact same syntax to wrap the code.'''
 
 
-template_generate_executor = PromptTemplate.from_template(
+gpt_35_turbo_usage_string = '''If you use gpt_3_5_turbo_api, then you must use in the following way:
+```
+from .apis import GPT_3_5_Turbo_API
+
+gpt_3_5_turbo_api = GPT_3_5_Turbo_API(system=SYSTEM_MESSAGE)  # fill-in the system message (str)
+gpt_completion = gpt_3_5_turbo_api(PROMPT)  # fill-in the prompt (str); the output is a string
```
Comment (joschkabraun) [inline src/options/generate/templates_user.py:95]: incorporate example case from system message

File: src/options/generate/templates_user.py [hunk 5]
```diff
@@ -164,14 +144,18 @@ def __call__(self, prompt: str) -> str:
...
-Make sure to include pytest. 
-Make sure to include openai>=0.26.0.
-Make sure that jina==3.15.1.dev14. 
-Make sure that docarray==0.21.0.
-You must not add gpt_3_5_turbo_api to the requirements.txt file.
-
-All versions are fixed using ~=, ==, <, >, <=, >=. The package versions must not have conflicts.
+Write the content of the requirements.txt file.
+The requirements.txt file must include the following packages:
+```
```
[the remark this answers is not in the document]
Comment (joschkabraun) [reply src/options/generate/templates_user.py:156]: works fine

File: src/options/generate/generator.py [hunk 6]
```diff
@@ -192,10 +200,10 @@ def generate_playground(self, microservice_name, microservice_path):
         print_colored('', '\n\n############# Playground #############', 'blue')
 
         file_name_to_content = get_all_microservice_files_with_content(microservice_path)
-        conversation = self.gpt_session.get_conversation([])
+        conversation = self.gpt_session.get_conversation(None)
```
Comment (florian-hoenicke) [inline src/options/generate/generator.py:203]: is it possible to not have none here?

**Review by florian-hoenicke — APPROVED** [review]

File: src/options/generate/generator.py (hunk 6 above)
Comment (joschkabraun) [reply src/options/generate/generator.py:203]: yes, shouldn't that be facilitated with your changes?

**Pull request merged**